library/core/src/main/java/com/google/android/exoplayer2/Format.java

```diff
@@ -150,6 +150,12 @@
    */
   public final long subsampleOffsetUs;
 
+  /**
+   * Boolean to indicate if the video is wide aspect ratio (16:9) or not (4:3)
+   * Only influences captions if the lines are not middle aligned.
+   * Defaulting to true as sources not having any flags are most probably wide screen
```

Comment:
CEA spec only indicates if the aspect ratio is for wide screen content or for letter boxed content, does not specify the actual ratio. A widescreen could mean any ratio greater than 1.55 i.e mid way of 1.33 (4:3) to 1.77 (16:9). 
floating point comparisons are tricky. 

https://stackoverflow.com/questions/1088216/whats-wrong-with-using-to-compare-floats-in-java

Reply:
Ideally, Format in its current form needs to be re-designed anyways because there are many fields that are applicable only to one type of format like Video or audio. In fact even language is applicable only to either audio or text.  

We should re-define Format that just has mimeType and perhaps Id,  and then have derived classes like VideoFormat, AudioFormat, TextFormat etc and have specific fields in each of them. Alternatively, modify initializationData to have a list of key/value pairs to make it easily extendable

I don't know if wide and non-wide captions would be applicable for other subtitle formats or not. I assumed usage of initializationData is like a binary blob that is sent to decoder which is not interpreted at app layer. But by using it in the way you advised, we are now giving meaning to the data elements of the blob. 

So here's what I'm gonna try to do. Introduce a generic hashtable to pass custom parameters. In future you may want to use it to remove accessibilityChannel and any other fields and make the Format Constructor light weight.

Reply:
> I assumed usage of initializationData is like a binary blob that is sent to decoder which is not interpreted at app layer.

It is a binary blob that's sent to the decoder. Where the decoder lives (app layer or platform) is not well defined. Even for audio and video, you can have decoders in both the platform layer (MediaCodec) and the app layer (e.g. the VP9 and FFMPEG extensions). For text the decoders all happen to be in the app layer.

> But by using it in the way you advised, we are now giving meaning to the data elements of the blob.

The blobs have a well defined structure in all cases (else whatever receives it wont know what to do with it). The only distinction is whether the app layer needs to know what the structure is, which it doesn't if it can transparently pass through the blob from the container to the platform in all cases, and which it does if it ever needs to construct or parse one itself.

Defining what we want the blob to look like for CEA formats seems reasonable to me. If it makes it better you can add a `CeaInitializationData` class in the `text.cea` package with methods for converting to/from a `byte[]`, and use that so that the structure of the blob is encapsulated within that class.

> So here's what I'm gonna try to do. Introduce a generic hashtable to pass custom parameters. In future you may want to use it to remove accessibilityChannel and any other fields and make the Format Constructor light weight.

Please don't do this. I agree that `Format` probably needs to be redesigned at some point, but for now `initializationData` is the mechanism for carrying this type of data. Adding a new alternative way of doing things makes things even more confusing.